aten/src/ATen/native/cuda/AmpKernels.cu

```diff
@@ -0,0 +1,110 @@
+#define _USE_MATH_DEFINES
+
+#include <math.h>
+
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cuda/Loops.cuh>
+
+
+namespace at {
+namespace native {
+
+// Multiplies scaled_grad in-place by inv_scale.  If an element of scaled_grad was inf or NaN sets found_inf to 1.0.
+//
+// Args:
+// scaled_grad:  A (scaled) gradient tensor.  May contain infs or NaNs.
+// found_inf:  A single-element float tensor to which 1.0 will be written if any gradients contain infs/nans.
+//             Pre-zeroing found_inf, if appropriate, is the responsibility of the caller.
+// inv_scale:  The inverse of the scale factor by which scaled_grad is currently multiplied.
+//
+// Returns:
+// A tuple with references to scaled_grad, which is now unscaled in place, and found_inf,
+// which is now guaranteed to contain 1.0 if an inf or NaN was found in scaled_grad.
+std::tuple<Tensor&, Tensor&> _amp_non_finite_check_and_unscale_cuda(Tensor& scaled_grad,
```

Comment:
Name should end in underscore. "found_inf" -> "found_non_finite?"

Greg suggests: "_amp_cuda_non_finite_check_and_unscale_"